Source/WebCore/dom/Document.cpp

```diff
@@ -3530,9 +3530,45 @@ void Document::setURL(const URL& url)
     m_url = WTFMove(newURL);
 
     m_documentURI = m_url.url().string();
+    m_adjustedURL = adjustedURL();
     updateBaseURL();
 }
 
+const URL& Document::urlForBindings() const
+{
+    auto shouldAdjustURL = [this] {
+        if (m_url.url().isEmpty())
+            return false;
+
+        if (!frame()->isMainFrame())
```

Comment:
I suspect you're missing a null check here. Or maybe instead, this could check if we're equal to the top document?